yb-voyager/cmd/config.go

```diff
@@ -375,57 +384,55 @@ func bindCobraFlagsToViper(cmd *cobra.Command, v *viper.Viper) error {
 	configKeyPrefix = setToAliasPrefixIfSet(configKeyPrefix, v)
 
 	cmd.Flags().VisitAll(func(f *pflag.Flag) {
-		if bindErr != nil {
-			return // Stop further execution if an error is already encountered
+		if bindErr != nil || f.Changed {
+			return // Skip already-set flags or if an error occurred
 		}
 
-		if !f.Changed {
-			// Check for <command_path>.<flagname>
-			if v.IsSet(configKeyPrefix + "." + f.Name) {
-				val := v.GetString(configKeyPrefix + "." + f.Name)
-				err := cmd.Flags().Set(f.Name, fmt.Sprintf("%v", val))
-				if err != nil {
-					// In case of an error while setting the flag from viper, return the error
-					bindErr = err
-					return
-				}
-			} else if v.IsSet(f.Name) {
-				// Bind the global flag from viper to cmd
-				val := v.GetString(f.Name)
-				err := cmd.Flags().Set(f.Name, fmt.Sprintf("%v", val))
-				if err != nil {
-					bindErr = err
-					return
-				}
-			} else if strings.HasPrefix(f.Name, "source-") && v.IsSet("source."+strings.TrimPrefix(f.Name, "source-")) {
-				// Handle source db type flags
-				val := v.GetString("source." + strings.TrimPrefix(f.Name, "source-"))
-				err := cmd.Flags().Set(f.Name, fmt.Sprintf("%v", val))
-				if err != nil {
-					bindErr = err
-					return
-				}
-			} else if strings.HasPrefix(f.Name, "oracle-") && v.IsSet("source."+f.Name) {
-				// Handle oracle db type flags, since they are also prefixed with source but are special cases
-				val := v.GetString("source." + f.Name)
-				err := cmd.Flags().Set(f.Name, fmt.Sprintf("%v", val))
-				if err != nil {
-					bindErr = err
-					return
-				}
-			} else if strings.HasPrefix(f.Name, "target-") && v.IsSet("target."+strings.TrimPrefix(f.Name, "target-")) {
-				// Handle target db type flags
-				val := v.GetString("target." + strings.TrimPrefix(f.Name, "target-"))
-				err := cmd.Flags().Set(f.Name, fmt.Sprintf("%v", val))
-				if err != nil {
-					bindErr = err
-					return
-				}
-				// fmt.Printf("binding %s from viper to cmd flag: %s=%s\n", "target."+strings.TrimPrefix(f.Name, "target-"), f.Name, val)
+		// Check for <command_path>.<flagname>
+		if v.IsSet(configKeyPrefix + "." + f.Name) {
+			val := v.GetString(configKeyPrefix + "." + f.Name)
+			err := cmd.Flags().Set(f.Name, val)
+			if err != nil {
+				// In case of an error while setting the flag from viper, return the error
+				bindErr = err
+				return
+			}
+		} else if v.IsSet(f.Name) {
+			// Bind the global flag from viper to cmd
+			val := v.GetString(f.Name)
+			err := cmd.Flags().Set(f.Name, val)
+			if err != nil {
+				bindErr = err
+				return
+			}
+		} else if strings.HasPrefix(f.Name, "source-") && v.IsSet("source."+strings.TrimPrefix(f.Name, "source-")) {
```

Comment:
let's use some constants like 
`SOURCE_DB_CONN_FLAG_PREFIX` , `SOURCE_DB_CONFIG_KEY_PREFIX` ,etc for `source-`, `source.`, `target-`, `target.`, `oracle-`, etc... 

Just helps to understand what the hyphen and dot are.

Reply:
Done